src/main/java/com/google/api/codegen/gapic/SurfaceGapicProvider.java

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.gapic;
+
+import com.google.api.codegen.transformer.ModelToSurfaceTransformer;
+import com.google.api.codegen.viewmodel.ViewModelDoc;
+import com.google.api.codegen.viewmodel.CommonSnippetSetRunner;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.stages.Merged;
+import com.google.api.tools.framework.snippet.Doc;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class SurfaceGapicProvider implements GapicProvider<Interface> {
```

Comment:
I fought with it a little bit but gave up. The issue is that I don't want getters for the fields in the main class, because they are only used internally, but using AutoValue for the builder forces you to make getters for them in the main class.